lib/ansible/modules/cloud/docker/docker_service.py

```diff
@@ -511,18 +511,17 @@ def make_redirection_tempfiles():
 
 
 def cleanup_redirection_tempfiles(out_name, err_name):
-    get_redirected_output(out_name)
-    get_redirected_output(err_name)
+    for i in [out_name, err_name]:
+        os.remove(i)
 
 
 def get_redirected_output(path_name):
     output = []
     with open(path_name, 'r') as fd:
         for line in fd:
             # strip terminal format/color chars
-            new_line = re.sub(r'\x1b\[.+m', '', line.encode('ascii'))
+            new_line = re.sub(r'\x1b\[.+m', '', to_text(line))
```

Comment:
This portion is wrong.  On python3 it doesn't do anything (because line is already text).  On python2 it makes a byte string pattern compare to a text string.  If there's no nonascii characters I think that will work okay but it will definitely fail to do the right thing if nonascii does creep into the data.